src/Rules.py

```diff
@@ -63,7 +64,13 @@ def set_rules(base: World, world: MultiWorld, player: int):
     # this is only called when the area (think, location or region) has a "requires" field that is a string
     def checkRequireStringForArea(state, area):
         requires_list = area["requires"]
-
+        
+        if not hasattr(state, 'manual_item_counts'): #Add a dict so this is only computed once per player, otherwise for a 2 player generation it add 25 seconds of proccess
+            state.manual_item_counts = {}
+        if player not in state.manual_item_counts:
+            real_pool = world.get_items()
+            state.manual_item_counts[player] = {i.name:real_pool.count(i) for i in real_pool if i.player == player}
+        items_counts = state.manual_item_counts[player]
```

Comment:
this and the 5ish lines above are the important part so generating doesnt take way too long